docker/ubuntu-14.04-cpu-all-options/Dockerfile

```diff
@@ -0,0 +1,72 @@
+FROM ubuntu:14.04
+MAINTAINER Aaron Markham <aaronmarkham@fb.com>
+
+# caffe2 install with cpu support
+
+########## REQUIRED DEPENDENCIES ################
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python-dev \
+    python-pip \
+    git \
+    build-essential \
+    cmake \
+    libprotobuf-dev \
+    protobuf-compiler \
+    libgoogle-glog-dev
+
+RUN git clone --recursive https://github.com/caffe2/caffe2.git && \
+    pip install --upgrade pip setuptools wheel && \
+    pip install future hypothesis numpy protobuf six
+
+########## OPTIONAL DEPENDENCIES ################
+
+RUN apt-get install -y --no-install-recommends \
+    libgflags-dev \
+    libgoogle-glog-dev \
+    libgflags2 \
+    libgtest-dev \
+    libiomp-dev \
+    libleveldb-dev \
+    liblmdb-dev \
+    libopencv-dev \
+    libopenmpi-dev \
+    libsnappy-dev \
+    openmpi-bin \
+    openmpi-doc \
+    python-numpy \
+    python-pydot \
+    python-setuptools \
+    python-scipy \
+    wget
+
+RUN pip install \
```

Comment:
Consider using `--no-cache-dir` to avoid bloating your image with cached wheels.